NDTensors/test/test_blocksparse.jl

```diff
@@ -1,22 +1,29 @@
+@eval module $(gensym())
 using NDTensors
-using LinearAlgebra
-using Test
+using LinearAlgebra: exp, Hermitian, svd
+using Test: @testset, @test, @test_throws
 using GPUArraysCore: @allowscalar
+include("NDTensorsTestUtils/NDTensorsTestUtils.jl")
+using .NDTensorsTestUtils: NDTensorsTestUtils
 
 @testset "BlockSparseTensor basic functionality" begin
   C = nothing
-  include("device_list.jl")
-  devs = devices_list(copy(ARGS))
 
-  @testset "test device: $dev" for dev in devs
-    elt = (dev == NDTensors.mtl ? Float32 : Float64)
+  @testset "test device: $dev, eltype: $elt" for dev in NDTensorsTestUtils.devices_list(
+      copy(ARGS)
+    ),
+    elt in (Float32, Float64)
+
+    if dev == NDTensors.mtl && elt == Float64
```

Comment:
TODO leaving a comment here to use the, is_supported_eltype function here and in other locations in testing